File: net/lib/schemas/NumberSchema.cs
using Albert.Lib.Rules;

namespace Albert.Lib.Schemas;

public class NumberSchema : BaseSchema<int>
{
    public override string Type => "string";

    public NumberSchema Min(int min)
    {
        this.AddRule(new NumberMinRule(min));
        return this;
    }

    public NumberSchema Max(int max)
    {
        this.AddRule(new NumberMaxRule(max));
        return this;
    }
}

File: js/lib/schemas/NumberSchema.ts
import { NumberMaxRule, NumberMinRule } from '../rules';
import { BaseSchema } from './BaseSchema';

export class NumberSchema extends BaseSchema<number> {
  constructor() {
    super('number');
  }

  min(min: number) {
    return this.addRule(new NumberMinRule(min));
  }

  max(max: number) {
    return this.addRule(new NumberMaxRule(max));
  }
}
